#895: added android-studio command

jan-vcapgemini opened the pull request:
Addresses: #895

Implements:
* first version of android-studio command for installation and application of settings

hohwille reviewed (approved):
@jan-vcapgemini thanks for your PR. You did an excellent job and this seems fully functional. 👍 
Great that you even included an integration tests.
Are you aware that we have fully functional integration tests here:
https://github.com/devonfw/ide/tree/master/scripts/src/test/bash

Further to test more of the actual code, I would recommend that `AndroidStudioUrlUpdaterMock` extends `AndroidStudioUrlUpdater`. Therefore you need to add a protected `getDownloadBaseUrl()` method returning `DOWNLOAD_URL` that can be overridden in the mock. Otherwise, to be honest you are not really testing the behavior of this feature as you can add any bug to `AndroidStudioUrlUpdater` as this is not tested at all but just your Mock.

I left some comments for documentation improvement for rework.
However, to signal that this is great work and functional, I review as `approved` as this could already be merged as is. However, I will leave it open so you can do some final polishing.

jan-vcapgemini commented on documentation/android-studio.asciidoc:
```diff
@@ -0,0 +1,54 @@
+
+= android-studio
+
+The `android-studio` commandlet allows to install, configure, and launch https://www.jetbrains.com/idea/[IntelliJ].
```
Thanks for pointing that out, fixed now.

jan-vcapgemini commented on documentation/android-studio.asciidoc:
```diff
@@ -0,0 +1,54 @@
+* `plugin_id` defines the unique ID of the plugin to install. If you want to customize `devonfw-ide` with new plugins use the search on https://plugins.jetbrains.com/androidstudio to find the plugin of your choice. Select the tab `Versions` and click on a version in the list. The plugin ID is displayed in the upper right corner. Copy & paste the ID from here to make up your own custom config.
+* `plugin_active` is an optional parameter. If it is `true` (default) the plugin will be installed automatically during the project link:setup.asciidoc[setup] for all developers in your team. Otherwise developers can still install the plugin manually via `devon android-studio add-plugin «plugin_id»`.
+
+In general you should try to stick with the configuration pre-defined by your project. But some plugins may be considered as personal flavor and are typically not predefined by the project config. Such plugins should be shipped with your link:settings.asciidoc[settings] as described above with `plugin_active=false` allowing you to easily install it manually. Surely, you can easily add plugins via ...
```
Thanks for pointing that out, fixed now.

jan-vcapgemini commented:
> @jan-vcapgemini thanks for your PR. You did an excellent job and this seems fully functional. 👍 Great that you even included an integration tests. Are you aware that we have fully functional integration tests here: https://github.com/devonfw/ide/tree/master/scripts/src/test/bash
> 
> Further to test more of the actual code, I would recommend that `AndroidStudioUrlUpdaterMock` extends `AndroidStudioUrlUpdater`. Therefore you need to add a protected `getDownloadBaseUrl()` method returning `DOWNLOAD_URL` that can be overridden in the mock. Otherwise, to be honest you are not really testing the behavior of this feature as you can add any bug to `AndroidStudioUrlUpdater` as this is not tested at all but just your Mock.
> 
> I left some comments for documentation improvement for rework. However, to signal that this is great work and functional, I review as `approved` as this could already be merged as is. However, I will leave it open so you can do some final polishing.

Thank you for your suggestions regarding the changes on the integration test. I've adjusted the test now.
About the mentioned integration tests using bash, I've added an integration test for Android Studio too already. I just hope that my java integration test can be used as a first example demonstrating WireMock and resource friendly tests.

hohwille reviewed (approved):
@jan-vcapgemini thanks for your rework 👍 Now everything is perfect.
You even did tidy up intellij docs to align with our conventions. Also the integration test now is great. Ready for merge.
